assets/js/googlesitekit/modules/datastore/settings.test.js

```diff
@@ -0,0 +1,123 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Modules from 'googlesitekit-modules';
+import { STORE_NAME } from './constants';
+import settings from './settings';
+import { createTestRegistry } from '../../../../../tests/js/utils';
+
+describe( 'core/modules module settings panel', () => {
+	let registry;
+	let store;
+	const slug = 'test-module';
+	const nonExistentModuleSlug = 'not-module';
+	const moduleStoreName = `modules/${ slug }`;
+
+	beforeEach( () => {
+		const storeDefinition = Modules.createModuleStore( moduleStoreName );
+		const isModuleActive = ( state, moduleSlug ) => {
+			return slug === moduleSlug ? true : null;
+		};
+
+		registry = createTestRegistry();
+		store = registry.stores[ STORE_NAME ].store;
+		registry.registerStore( STORE_NAME, Data.combineStores(
+			store,
+			{
+				selectors: {
+					isModuleActive,
+				},
+			},
+			settings,
+		) );
+
+		registry.registerStore( moduleStoreName, Data.combineStores(
+			storeDefinition,
+		) );
+	} );
+
+	describe( 'reducer', () => {
+		it( 'has the appropriate initial state', () => {
+			expect( true ).toBeTruthy();
+			const initialState = {};
+			const initialStoreState = store.getState();
+
+			// Ensure we have default state correct
+			expect( Object.keys( initialStoreState ) ).toContain( 'panelState' );
+			expect( initialStoreState.panelState ).toEqual( initialState );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getModuleSettingsPanelState', () => {
+			it( 'gets module settings panel state', () => {
+				// Default state for valid module is closed
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'closed' );
+				// Default state for invalid module is closed
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( nonExistentModuleSlug ) ).toBe( null );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'view' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'view' );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'closed' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'closed' );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'edit' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'edit' );
```

Comment:
These tests lack a bit of context as to why they are relevant (allowed values) but that is missing here.

I would say these actually belong on the tests for the action since that is where the restriction is; the selector basically just grabs the value from state. That way we can keep the tests for action and selector separate as much as possible, although in some cases it isn't possible to do so.